charts/spiderpool/templates/pod.yaml

```diff
@@ -1,159 +1,114 @@
-    - name: SPIDERPOOL_INIT_DEFAULT_IPV4_IPPOOL_NAME
-      value: {{ .Values.clusterDefaultPool.ipv4IPPoolName | quote }}
-    - name: SPIDERPOOL_INIT_DEFAULT_IPV4_IPPOOL_SUBNET
-      value: {{ .Values.clusterDefaultPool.ipv4Subnet | quote }}
-    - name: SPIDERPOOL_INIT_DEFAULT_IPV4_IPPOOL_IPRANGES
-      value: {{ toJson .Values.clusterDefaultPool.ipv4IPRanges | quote }}
-    - name: SPIDERPOOL_INIT_DEFAULT_IPV4_IPPOOL_GATEWAY
-      value: {{ .Values.clusterDefaultPool.ipv4Gateway | quote }}
-    {{- if .Values.ipam.spidersubnet.enable }}
-    - name: SPIDERPOOL_INIT_DEFAULT_IPV4_SUBNET_NAME
-      value: {{ .Values.clusterDefaultPool.ipv4SubnetName | quote }}
-    {{- end }}
-    {{- end }}
-    {{- if and .Values.clusterDefaultPool.installIPv6IPPool .Values.ipam.enableIPv6 }}
-    - name: SPIDERPOOL_INIT_DEFAULT_IPV6_IPPOOL_NAME
-      value: {{ .Values.clusterDefaultPool.ipv6IPPoolName | quote }}
-    - name: SPIDERPOOL_INIT_DEFAULT_IPV6_IPPOOL_SUBNET
-      value: {{ .Values.clusterDefaultPool.ipv6Subnet | quote }}
-    - name: SPIDERPOOL_INIT_DEFAULT_IPV6_IPPOOL_IPRANGES
-      value: {{ toJson .Values.clusterDefaultPool.ipv6IPRanges | quote }}
-    - name: SPIDERPOOL_INIT_DEFAULT_IPV6_IPPOOL_GATEWAY
-      value: {{ .Values.clusterDefaultPool.ipv6Gateway | quote }}
-    {{- if .Values.ipam.spidersubnet.enable }}
-    - name: SPIDERPOOL_INIT_DEFAULT_IPV6_SUBNET_NAME
-      value: {{ .Values.clusterDefaultPool.ipv6SubnetName | quote }}
-    {{- end }}
-    {{- end }}
-    - name: SPIDERPOOL_INIT_ENABLE_MULTUS_CONFIG
-      value: {{ .Values.multus.enableMultusConfig | quote }}
-    - name: SPIDERPOOL_INIT_INSTALL_MULTUS
-      value: {{ .Values.multus.multusCNI.install | quote }}
-    - name: SPIDERPOOL_INIT_DEFAULT_CNI_NAME
-      value: {{ .Values.multus.multusCNI.defaultCniCRName | quote }}
-    - name: SPIDERPOOL_INIT_DEFAULT_CNI_NAMESPACE
-      value: {{ .Release.Namespace | quote }}
-    - name: SPIDERPOOL_INIT_MULTUS_CONFIGMAP
-      value: {{ .Values.multus.multusCNI.name | trunc 63 | trimSuffix "-" | quote }}
-    {{- if eq .Values.multus.multusCNI.defaultCniCRName "" }}
-    - name: SPIDERPOOL_INIT_DEFAULT_CNI_DIR
-      value: {{ .Values.global.cniConfHostPath | quote }}
-    - name: SPIDERPOOL_INIT_READINESS_FILE
-      value: "/etc/spiderpool/ready"
-    volumeMounts:
-      - name: cni
-        mountPath: {{ .Values.global.cniConfHostPath }}
-  volumes:
-    - name: cni
-      hostPath:
-        path: {{ .Values.global.cniConfHostPath }}
-    {{- end }}
+          - {{ .Values.spiderpoolInit.binName }}
+        {{- with .Values.spiderpoolInit.extraArgs }}
+        args:
+        {{- toYaml . | trim | nindent 4 }}
+        {{- end }}
+        readinessProbe:
+          exec:
+            command:
+            - cat
+            - "/etc/spiderpool/ready"
+          initialDelaySeconds: 15
+          periodSeconds: {{ .Values.spiderpoolInit.readinessProbe.periodSeconds }}
```

Comment:
the status of pod indicate the installation. So, I am wondering it is neccerary to have the readiness and liveness

Reply:
readinessProbe is make sense, because we need to wait for the spiderpool-agent to be ready before we can use it.